### PR TITLE
Will add an option in the config to generate a token for authentication for github

### PR DESCRIPTION
**Pull request by Thraxis**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by fernandog** [comment]
tested and all good! nice job

**Comment by fernandog** [comment]
rebased and fixed conflict

**Comment by fernandog** [comment]
I would add a note in modal saying that user dont need to change any settings. He only needs to generate the token

File: static/js/common/init.js [hunk 1]
```diff
@@ -2,6 +2,8 @@ MEDUSA.common.init = function() {
     // Import underscore.string using it's mixin export.
     _.mixin(s.exports());
 
+    var common = {}
```
Comment (OmgImAlexis) [inline static/js/common/init.js]: ?

File: static/js/common/init.js [hunk 2]
```diff
@@ -24,6 +26,45 @@ MEDUSA.common.init = function() {
...
+    // function to change luminance of #000000 color - used in triggerhighlighting
+    function ColorLuminance(hex, lum) {
+        hex = String(hex).replace(/[^0-9a-f]/gi, '');
+        if (hex.length < 6) {
+            hex = hex[0]+hex[0]+hex[1]+hex[1]+hex[2]+hex[2];
+        }
+        lum = lum || 0;
+        var rgb = "#", c, i;
+        for (i = 0; i < 3; i++) {
+            c = parseInt(hex.substr(i*2,2), 16);
```
Comment (OmgImAlexis) [inline static/js/common/init.js]: This needs a radix.

File: static/js/common/init.js [hunk 3]
```diff
@@ -24,6 +26,45 @@ MEDUSA.common.init = function() {
         return offset;
     }
 
+    // function to change luminance of #000000 color - used in triggerhighlighting
+    function ColorLuminance(hex, lum) {
```
Comment (OmgImAlexis) [inline static/js/common/init.js]: Function names start in lowercase.

File: static/js/common/init.js [hunk 4]
```diff
@@ -24,6 +26,45 @@ MEDUSA.common.init = function() {
...
+    // function to convert rgb(0,0,0) into #000000
+    function rgb2hex(rgb) {
+        rgb = rgb.match(/^rgb\((\d+),\s*(\d+),\s*(\d+)\)$/);
+        function hex(x) {
+            return ("0" + parseInt(x).toString(16)).slice(-2);
+        }
+        return "#" + hex(rgb[1]) + hex(rgb[2]) + hex(rgb[3]);
+    }
+
+    var revert_background_color; // used to revert back to original background-color after highlight
```
Comment (OmgImAlexis) [inline static/js/common/init.js]: camelCase everything please.

File: static/js/common/init.js [hunk 5]
```diff
@@ -24,6 +26,45 @@ MEDUSA.common.init = function() {
...
+            rgb += ("00"+c).substr(c.length);
+        }
+        return rgb;
+    }
+
+    // function to convert rgb(0,0,0) into #000000
+    function rgb2hex(rgb) {
+        rgb = rgb.match(/^rgb\((\d+),\s*(\d+),\s*(\d+)\)$/);
+        function hex(x) {
+            return ("0" + parseInt(x).toString(16)).slice(-2);
```
Comment (OmgImAlexis) [inline static/js/common/init.js]: radix needed here too.

File: static/js/common/init.js [hunk 6]
```diff
@@ -24,6 +26,45 @@ MEDUSA.common.init = function() {
...
+    var allCells = $(".triggerhighlight");
+    allCells
+    .on("mouseover", function() {
+        var el = $(this),
+          pos = el.index();
+        revert_background_color = rgb2hex($(this).parent().css("background-color")); // fetch the original background-color to revert back to
+        var highlight_background_color = ColorLuminance(revert_background_color, -0.15); // change highlight color based on original color
+        el.parent().find(".triggerhighlight").css("background-color", highlight_background_color); // setting highlight background-color
+    })
+    .on("mouseout", function() {
```
Comment (OmgImAlexis) [inline static/js/common/init.js]: Should be on the line before as `}).on("mouseout", function() {`

File: static/js/common/init.js [hunk 7]
```diff
@@ -24,6 +26,45 @@ MEDUSA.common.init = function() {
...
+    function rgb2hex(rgb) {
+        rgb = rgb.match(/^rgb\((\d+),\s*(\d+),\s*(\d+)\)$/);
+        function hex(x) {
+            return ("0" + parseInt(x).toString(16)).slice(-2);
+        }
+        return "#" + hex(rgb[1]) + hex(rgb[2]) + hex(rgb[3]);
+    }
+
+    var revert_background_color; // used to revert back to original background-color after highlight
+    var allCells = $(".triggerhighlight");
```
Comment (OmgImAlexis) [inline static/js/common/init.js]: Why make a var if it's only used once? Just use the on with the jQuery selector.

File: static/js/common/init.js (hunk 6 above)
Comment (OmgImAlexis) [inline static/js/common/init.js]: Single quotes please.

File: static/js/config/init.js [hunk 8]
```diff
@@ -112,4 +112,39 @@ MEDUSA.config.init = function() {
...
     });
+
+    // GitHub Auth Types
+    setupGithubAuthTypes();
+
+    $('input[name="git_auth_type"]').on('click', function() {
+        setupGithubAuthTypes();
+    });
+
+    function setupGithubAuthTypes() {
```
Comment (OmgImAlexis) [inline static/js/config/init.js]: A function is to be added BEFORE it's called. We can't call before it exists.

File: static/js/config/init.js [hunk 9]
```diff
@@ -112,4 +112,39 @@ MEDUSA.config.init = function() {
...
+        var selected = parseInt($('input[name="git_auth_type"]').filter(':checked').val());
```
Comment (OmgImAlexis) [inline static/js/config/init.js]: Why not use a `data` attribute as they keep ints as ints.

**Comment by labrys** [comment]
@Thraxis is there a reason for making this a modal when the rest of the configurations like this (e.g. notifications, providers, etc) are all modeless?

**Comment by Thraxis** [comment]
Replaced the modal dialog with a popover:

image

File: static/js/common/init.js [hunk 10]
```diff
@@ -56,8 +56,7 @@ MEDUSA.common.init = function() {
     }
 
     var revertBackgroundColor; // used to revert back to original background-color after highlight
-    var allCells = $('.triggerhighlight');
-    allCells.on('mouseover', function() {
+    $('.triggerhighlight').on('mouseover', function() {
         var el = $(this);
```
Comment (OmgImAlexis) [inline static/js/common/init.js]: Again, if it's only used once you might as well just use the selector.

Comment (Thraxis) [inline static/js/common/init.js]: Umm... pretty sure I am using the selector there

**Pull request merged**